### ISSUE TITLE
Controls with bottom anchor are inexplicably shrunk in PerMonitor/V2 even with AutoScaleMode.None

### ISSUE THREAD

**Issue by jnm2** [dotnet/winforms#10490]
### .NET version

Issue is the same on both .NET 6.0 and 8.0.

### Did it work in .NET Framework?

Not tested/verified

### Issue description

When the bottom edge of the label gets near the padding of the bottom edge of the containing form, the label gets inexplicably resized at runtime.

Designer:
![image](https://github.com/dotnet/winforms/assets/8040367/6e479e02-3810-4363-b192-b1f2684145e1)

Running the app:
![image](https://github.com/dotnet/winforms/assets/8040367/fcadb2b1-f57f-4923-b270-9d0f57598efb)

### Steps to reproduce

[Repro.zip](https://github.com/dotnet/winforms/files/13694666/Repro.zip)

InitializeComponent contents from the full attached repro:

```cs
label1 = new System.Windows.Forms.Label();
SuspendLayout();
// 
// label1
// 
label1.Anchor = System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Left | System.Windows.Forms.AnchorStyles.Right;
label1.BackColor = System.Drawing.Color.Fuchsia;
label1.Location = new System.Drawing.Point(3, 87);
label1.Name = "label1";
label1.Size = new System.Drawing.Size(380, 24);
label1.TabIndex = 1;
label1.Text = "Distance to the bottom of the form is important";
label1.TextAlign = System.Drawing.ContentAlignment.BottomCenter;
// 
// MainForm
// 
AutoScaleMode = System.Windows.Forms.AutoScaleMode.None;
ClientSize = new System.Drawing.Size(386, 126);
Controls.Add(label1);
Name = "MainForm";
Padding = new System.Windows.Forms.Padding(0, 0, 0, 16);
Text = "Windows Forms repro";
ResumeLayout(false);

**Comment by elachlan** [comment]
@jnm2 Does it work as expected when you set  "System.Windows.Forms.AnchorLayoutV2": true in the runtimeconfig.json?

Possibly related: #10382

**Comment by elachlan** [comment]
@Olina-Zhang can your team please test?

**Comment by LeafShi1** [comment]
> When the bottom edge of the label gets near the padding of the bottom edge of the containing form, the label gets inexplicably resized at runtime.

Yes, set "System.Windows.Forms.AnchorLayoutV2": true will resolve this issue.

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang can your team please test?

From provided sample project, we can see the label shrinks in High DPI environment, that is because of setting label's **AutoSize** with `false`. Once changing it to `true`, issue will be resolved.
<img width="532" alt="image" src="https://github.com/dotnet/winforms/assets/26474449/83ff8777-b232-469a-8777-3c2836beeec9">

https://github.com/dotnet/winforms/assets/26474449/fb3b81cf-9cb5-4561-9a51-2dfbb424134b

**Comment by jnm2** [comment]
AutoSize would not be an appropriate setting in the full app where the problem was seen. The label purposely fills space and puts text at the bottom.

**Comment by jnm2** [comment]
@elachlan Yes, this fixed the problem.

```json
{
  "configProperties": {
    "System.Windows.Forms.AnchorLayoutV2": true
  }
}
```
